CMakeLists.txt

```diff
@@ -111,6 +111,8 @@ foreach(_file ${_check_files})
   verbose_include(${_file})
 endforeach()
 
+verbose_include(${CMAKE_SOURCE_DIR}/cmake/setup_finalize.cmake)
```

Comment:
I think this is a refactoring remnant.

Even when moving the inclusion back to its original position it will no longer test the link interface - because everything now happens with interface targets. I have added a todo to refactor the `check_compiler_setup()` macro to do that again.

I suggest we leave this as is for the time being and retire/refactor the entire file in a follow up pull request when I address all of the open TODOs.

Reply:
On a related note: I will try to get rid of this quirky `*_SAVED` flag handling as well. if this is successful then there isn't really much happening in this file any more.